mito-ai/mito_ai/tests/providers_test.py

```diff
@@ -183,6 +192,169 @@ def test_gemini_model_resolution(monkeypatch: pytest.MonkeyPatch) -> None:
         resolved = llm._resolve_model("gpt-4o-mini")
         assert resolved == "gemini-2-pro"
 
+
+def test_gemini_capabilities(monkeypatch: pytest.MonkeyPatch) -> None:
+    """Test that Gemini capabilities are correctly reported when configured."""
+    # Mock both environment variables and constants
+    monkeypatch.setenv("GEMINI_API_KEY", "gemini-key")
+    monkeypatch.setenv("GEMINI_MODEL", "gemini-2-pro")
+    monkeypatch.setattr("mito_ai.constants.GEMINI_API_KEY", "gemini-key")
+    monkeypatch.setattr("mito_ai.constants.GEMINI_MODEL", "gemini-2-pro")
+    # Ensure other providers are not configured
+    monkeypatch.setattr("mito_ai.constants.OPENAI_API_KEY", None)
+    monkeypatch.setattr("mito_ai.constants.OLLAMA_MODEL", None)
+    monkeypatch.setattr("mito_ai.constants.CLAUDE_API_KEY", None)
+    monkeypatch.setattr("mito_ai.constants.CLAUDE_MODEL", None)
+    monkeypatch.setattr("mito_ai.enterprise.utils.is_azure_openai_configured", lambda: False)
+
+    llm = OpenAIProvider()
+    capabilities = llm.capabilities
+    assert capabilities.provider == "Gemini"
+    assert capabilities.configuration["model"] == "gemini-2-pro"
+
+
+def test_gemini_client_initialization(monkeypatch: pytest.MonkeyPatch) -> None:
+    """Test that Gemini client is properly initialized when configured."""
+    # Mock both environment variables and constants
+    monkeypatch.setenv("GEMINI_API_KEY", "gemini-key")
+    monkeypatch.setenv("GEMINI_MODEL", "gemini-2-pro")
+    monkeypatch.setattr("mito_ai.constants.GEMINI_API_KEY", "gemini-key")
+    monkeypatch.setattr("mito_ai.constants.GEMINI_MODEL", "gemini-2-pro")
+    # Ensure other providers are not configured
+    monkeypatch.setattr("mito_ai.constants.OPENAI_API_KEY", None)
+    monkeypatch.setattr("mito_ai.constants.OLLAMA_MODEL", None)
+    monkeypatch.setattr("mito_ai.constants.CLAUDE_API_KEY", None)
+    monkeypatch.setattr("mito_ai.constants.CLAUDE_MODEL", None)
+    monkeypatch.setattr("mito_ai.enterprise.utils.is_azure_openai_configured", lambda: False)
+
+    # Create a mock GeminiClient class
+    mock_gemini_class = MagicMock()
+    mock_instance = MagicMock()
+    mock_gemini_class.return_value = mock_instance
+    
+    with patch("mito_ai.providers.GeminiClient", mock_gemini_class):
+        llm = OpenAIProvider()
+        assert llm._gemini_client is not None
+        # Verify the client was initialized with correct parameters
+        mock_gemini_class.assert_called_once_with(
+            api_key="gemini-key",
+            model="gemini-2-pro"
+        )
+
+
+@pytest.mark.asyncio
+async def test_gemini_completion_request(monkeypatch: pytest.MonkeyPatch) -> None:
```

Comment:
It seems like we are mocking so much in this test, what are we actually testing?